gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsTimeStatistic.java

```diff
@@ -0,0 +1,104 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import java.util.EnumSet;
+import org.apache.hadoop.classification.InterfaceStability;
+
+/** Time Statistics which are collected in GCS. */
+@InterfaceStability.Unstable
+public enum GhfsTimeStatistic {
```

Comment:
I am not sure creating a mutable enum is the right thing to do. Why not create class instead? Also having this singleton make the testing harder

Reply:
For the plugin GhfsTimeStatisticPlugin, I had references to loop over various metrics. This is the reason why mutable enum was used. Also by class, do you mean an implementation like https://github.com/LucaCanali/hadoop/blob/s3aAndHDFSTimeInstrumentation/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ATimeInstrumentation.java ?

Reply:
I think mutable enums are an anti-pattern since it is sort of like global variables, which are hard to test etc. I was curious why can't we achieve the same functionality we need by using Java class instead of using Java Enum?

Reply:
Having enum class makes the metrics look more organized in my opinion. I agree with mutable enum issue so I have mapped them with AtomicLong and used it to store the metric's value. Does that seem like a good alternative ?

Reply:
This is practically a global mutable variable and testing this will be difficult. Hence I do not think this is the right thing to do.

Reply:
Please open a bug for this and assign to me. Also add a TODO comment in the code with the bug number.